app/assets/javascripts/discourse/controllers/bookmark.js

```diff
@@ -73,12 +75,34 @@ export default Controller.extend(ModalFunctionality, {
       userTimezone: this.currentUser.resolvedTimezone()
     });
 
+    this.parseExistingBookmarkReminderAt();
```

Comment:
There's a lot of methods with a a kinda unexpected naming, let me explain my reasoning.

When you say: `this.parseExistingBookmarkReminderAt();` I expect to actually parse something as a param, here the parsing is actually happening inside the function through some state and mutates another state.

I feel like a better separation of public/private APIs should be though here, as it currently makes the code a little bit hard to follow in my opinion.

Reply:
That makes sense, keeping mutations like this as pure functions. I had a moment of doubt yesterday when I made and named it too but just brushed past it. Will make this more clear :+1: